### PR TITLE
xpath metaparser based on libxml2

### PR DESCRIPTION
**Pull request by masatake**
ant.c can be rewritten with xpath metaparser.

### CODE REVIEW

**Comment by masatake** [comment]
Remove unnecessary white spaces..

**Comment by masatake** [comment]
For extension ".xml" we need a good selector.

File: parsers/glade.c [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+};
+
+static void
+findGladeTags (void)
+{
+	xmlDocPtr doc;
+	xmlNode *root;
+	xmlXPathContextPtr ctx;
+
+	findRegexTags ();
```
Comment (b4n) [inline parsers/glade.c]: ?

File: parsers/glade.c [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+	doc = xmlParseFile(getInputFileName());
```
Comment (b4n) [inline parsers/glade.c]: It's a bit sad to use the filename, as it makes it a little harder to move to in-memory parsing if we want to at some point.

File: parsers/glade.c [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+	ctx = xmlXPathNewContext (doc);
+	if (ctx == NULL)
+		error (FATAL, "failed to make a new xpath context for %s", getInputFileName());
+
+	root = xmlDocGetRootElement(doc);
+	if (root == NULL)
+	{
+		verbose ("could not get the root node for %s", getInputFileName());
+		goto out;
+	}
```
Comment (b4n) [inline parsers/glade.c]: won't all this initialization be the same in all Xpath parsers, and so can't it be moved to some generic place?

File: parsers/glade.c (hunk 1 above)
Comment (b4n) [inline parsers/glade.c]: oh, for the doctype extraction in the example.  sounds a bit hacky, couldn't this be extracted with an Xpath that could get installed at runtime?

File: parsers/glade.c (hunk 2 above)
Comment (masatake) [inline parsers/glade.c]: Oh, I see. However, we can improve this later.

File: parsers/glade.c (hunk 1 above)
Comment (masatake) [inline parsers/glade.c]: Drive the regex parser here.
See Units/simple-glade.d/args.ctags in these commits.

File: parsers/glade.c (hunk 3 above)
Comment (masatake) [inline parsers/glade.c]: The interface between glade.c (concrete xpath parser) and lxpath.c (meta xpath parser) is under designed. I have to write more concrete xpath parsers to find the good interface.

The redundancy is for introducing "--xpath-<LANG>=..." option in the future.
As you wrote, I have to refine the interface.

File: parsers/glade.c (hunk 1 above)
Comment (masatake) [inline parsers/glade.c]: It is just an example. What I want to say is that calling findRegexTags is meaningful; it allows a user doing something hacky.

File: parsers/glade.c (hunk 2 above)
Comment (masatake) [inline parsers/glade.c]: I found `xmlParseMemory'. I will try to use this.

**Comment by masatake** [comment]
Scope is added but the interface between a concrete parser and xpath meta parser becomes ugly.
```
 ./ctags -o - ~/var/ctags-github/Units/simple-dbusinspect.d/input.xml  | sed -e 's|/home/yamato/var/ctags-github/Units/simple-dbusinspect.d/||'
</Units/simple-dbusinspect.d/input.xml  | sed -e 's|/home/yamato/var/ctags-github/Units/simple-dbusinspect.d/||'
ClientAdded	input.xml	/^    <signal name="ClientAdded">$/;"	s	interface:org.gnome.SessionManager
SessionName	input.xml	/^    <property name="SessionName" type="s" access="read">$/;"	p	interface:org.gnome.SessionManager
Setenv	input.xml	/^    <method name="Setenv">$/;"	m	interface:org.gnome.SessionManager
org.gnome.SessionManager	input.xml	/^  <interface name="org.gnome.SessionManager">$/;"	i
```

It seems much more work is needed...

**Comment by masatake** [comment]
ugly and inefficient but I implemented generic selector for xml parser family. (2658334).
Now xpath meta parser can be used in real life. 
After working on ant.c I would like to merge.

@b4n gives me an homework: using xmlParseMemory instead of xmlParseFile.

**Comment by masatake** [comment]
The interface of xpath meta parser is updated as @b4n suggested.

**Pull request merged**